test/rdfTypes/VariableTest.cpp

```diff
@@ -30,3 +32,74 @@ TEST(Variable, DollarToQuestionMark) {
   EXPECT_EQ(x.name(), "?x");
   EXPECT_EQ(x2.name(), "?x");
 }
+
+// _____________________________________________________________________________
+TEST(Variable, ScoreAndMatchVariablesUnicode) {
+  ad_utility::HashSet<Variable> vars;
+  auto f = [&vars](Variable var) { vars.insert(std::move(var)); };
+
+  // Test that the following variables all have valid names (otherwise the
+  // constructor of `Variable` would throw), and that they are all unique.
+  EXPECT_NO_THROW(f(Variable("?x").getWordScoreVariable("\U0001F600", false)));
+  EXPECT_NO_THROW(f(Variable("?x").getWordScoreVariable("\U0001F600*", true)));
+  EXPECT_NO_THROW(f(Variable("?x").getWordScoreVariable("äpfel", false)));
+  EXPECT_NO_THROW(f(Variable("?x").getEntityScoreVariable("\U0001F600")));
+  EXPECT_NO_THROW(f(Variable("?x").getEntityScoreVariable("äpfel")));
+  EXPECT_NO_THROW(f(Variable("?x").getMatchingWordVariable("äpfel")));
+  EXPECT_NO_THROW(f(Variable("?x").getMatchingWordVariable("\U0001F600")));
+
+  // Examples of characters that are alphabetic according to Unicode, but are
+  // not valid in SPARQL variables.
+  EXPECT_NO_THROW(f(Variable("?x").getMatchingWordVariable("\U000000AA")));
+  EXPECT_NO_THROW(f(Variable("?x").getMatchingWordVariable("\U000000B5")));
+  EXPECT_NO_THROW(f(Variable("?x").getMatchingWordVariable("\U000000BA")));
+
+  // Expect that all the created variables are unique.
+  EXPECT_EQ(vars.size(), 10);
+
+  // Ensure that underscores which are used to escape unsupported code points
+  // are themselves escaped to ensure unique variable names.
+  EXPECT_EQ(Variable("?x").getMatchingWordVariable("_").name(),
+            "?ql_matchingword_x__95_");
+
```

Comment:
In the `legalAndIllegalVariableNames` test above we can now add emojis and asian scripts :)